loyalty_criteria_multi_product/migrations/16.0.1.0.0/pre-migration.py

```diff
@@ -19,10 +19,13 @@ def move_coupon_criteria_to_rule(env):
         env.cr,
         """
         UPDATE loyalty_rule AS lr
-        SET loyalty_criteria = 'multi_product'
+        SET loyalty_criteria = CASE
+            WHEN lp.coupon_criteria = 'multi_product' THEN 'multi_product'
+            WHEN lp.coupon_criteria = 'domain' THEN 'domain'
+        END
         FROM loyalty_program AS lp
         WHERE lr.program_id = lp.id
-        AND lp.coupon_criteria = 'multi_product'
+        AND lp.coupon_criteria IN ('multi_product', 'domain')
```

Comment:
Are you sure you need to do this restriction?

Reply:
You are right, it is not necessary because everyone will have an established criteria.